### PR TITLE
Troubleshooting: can't log in

### PR DESCRIPTION
**Pull request by gvwilson**
Troubleshooting login problems.

### CODE REVIEW

File: docs/troubleshooting-guide/cant-log-in.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
+# People can't log in to Metabase
+
+You are supposed to be able to log in to Metabase, but:
```
Comment (jeff-bruemmer) [inline docs/troubleshooting-guide/cant-log-in.md:3]:
```suggestion
You should be able to log in to Metabase, but:
```

File: docs/troubleshooting-guide/cant-log-in.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+- you can't see the login page, or
+- your credentials aren't accepted.
+
+## Are you using the right URL?
```
Comment (jeff-bruemmer) [inline docs/troubleshooting-guide/cant-log-in.md:8]:
```suggestion
## Are you using the right URL for your Metabase?
```

File: docs/troubleshooting-guide/cant-log-in.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+- Do you have the port as well as the host?
+- Has someone moved the Metabase instance (e.g., it was in dev and now it's in production)?
+
+## Are the credentials stored by Metabase or is the account managed using Single Sign-On?
```
Comment (jeff-bruemmer) [inline docs/troubleshooting-guide/cant-log-in.md:13]:
```suggestion
## Does your Metabase use Single Sign-On (SS0) to manage accounts?
```

File: docs/troubleshooting-guide/cant-log-in.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+- Other steps depend on knowing this
+- FIXME: can people look at their own profile to find out how their credentials are managed?
```
Comment (jeff-bruemmer) [inline docs/troubleshooting-guide/cant-log-in.md:16]: The tell here should be obvious when they sign in. They should have an option to use the third party identity provider, or "sign in with email".

File: docs/troubleshooting-guide/cant-log-in.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+- FIXME: where in the Admin Panel can someone find this?
+
+Note Metabase can't re-set passwords for SSO: if (for example) someone is logging in using their Google ID, they must re-set the Google password associated with that account.
```
Comment (jeff-bruemmer) [inline docs/troubleshooting-guide/cant-log-in.md:19]:
```suggestion
Metabase can't reset passwords for SSO: if (for example) someone is logging in using their Google ID, they must reset the Google password associated with that account.
```

File: docs/troubleshooting-guide/cant-log-in.md [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+## If the credentials are stored in Metabase, has the account been deactivated?
```
Comment (jeff-bruemmer) [inline docs/troubleshooting-guide/cant-log-in.md:21]:
```suggestion
## If they sign in by email (not by SSO), has their Metabase account been deactivated?
```

File: docs/troubleshooting-guide/cant-log-in.md [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+- If there is no "Deactivated" tab, there are no deactivated accounts
+- If there is a "Deactivated" tab, look for the user in question
+- Click on the "recycle" loop arrow to reactivate the account
+
+## If the credentials are stored in Metabase, are they using the correct ID and password?
+
+- [This FAQ][reset-password] explains how to re-set your own password
+- Admins can also go to Admin Panel > People > "..." > "Reset password"
+
+## If the account is managed by SSO, is SSO enabled?
```
Comment (jeff-bruemmer) [inline docs/troubleshooting-guide/cant-log-in.md:34]: We can probably skip this: if it's managed by SSO, then it's enabled.

**Review by jeff-bruemmer — COMMENTED** [review]
This pretty much covers the gamut, though I recommend splitting this into two sections: sign-in with email, and SSO so readers can skip to content relevant to their setup. We might also want to update https://www.metabase.com/docs/latest/faq/using-metabase/how-do-i-reset-my-password.html, as some people have been downvoting that article (likely because it doesn't cover SSO) or delete it and redirect here.

File: docs/troubleshooting-guide/cant-log-in.md [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+## Are you using the right URL for your Metabase?
+
+**Root cause:** the Metabase instance you are trying to log in to isn't where you think it is or isn't accessible.
+
+**Steps to take:**
+
+1. Check whether you need to include a port number as well as a hostname in the connection URL. For example, Metabase might be at `https://example.com:3000/` instead of `https://example.com/`.
+   - If you're an administrator, you'll have configured this.
+   - If you're not, please ask your admin.
+2. Check whether your Metabase instance has moved. For example, if you were using a trial instance of Metabase, but you're now in production, the URL might have changed.
```
Comment (flamber) [inline docs/troubleshooting-guide/cant-log-in.md:17]: I don't think I have seen more than a couple having this problem over the years.

File: docs/troubleshooting-guide/cant-log-in.md [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+
+## Does your Metabase use Single Sign-On (SS0) to manage accounts?
+
+Metabase can manage accounts itself, or administrators can configure it to let people sign on using third-party credentials. You can tell what kind of account you have when you try to sign in: if your login ID is your email address, Metabase is managing the account, but if you're using the option to sign in which a third-party identity provider, someone else is responsible for authenticating you.
```
Comment (flamber) [inline docs/troubleshooting-guide/cant-log-in.md:21]: I would remove this. It's also not correct - LDAP would be an example.

File: docs/troubleshooting-guide/cant-log-in.md [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+**Root cause:** You have forgotten your password.
+
+**Steps to take:**
+
+1. If you are using the desktop Mac App, click on the **Help** menu item and select `Reset Password`.
+2. If you are an administrator and want to reset someone's passsword, go to **Admin Panel**, select **People**, click on the ellipsis "..." next to the person's account, and select `Reset Password`.
+3. If you are using the web app as a normal user:
+   1. Click the link in the lower-right of the login screen that reads, "I seem to have forgotten my password".
+   2. If your Metabase administrator has [set up email][setting-up-email] you will receive a password reset email.
+   3. If email has not been configured, you will need to contact a Metabase admin to perform a password reset.
```
Comment (flamber) [inline docs/troubleshooting-guide/cant-log-in.md:49]: Would probably move this to the top - it's the most common problem.

**Review by flamber — APPROVED** [review]

File: docs/troubleshooting-guide/cant-log-in.md [hunk 11]
```diff
@@ -0,0 +1,95 @@
...
+1. Check whether you need to include a port number as well as a hostname in the connection URL. For example, Metabase might be at `https://example.com:3000/` instead of `https://example.com/`.
+   - If you're an administrator, you'll have configured this.
+   - If you're not, please ask your admin.
+2. Check whether your Metabase instance has moved. For example, if you were using a trial instance of Metabase, but you're now in production, the URL might have changed.
+
+## Does your Metabase use Single Sign-On (SS0) to manage accounts?
+
+Metabase can manage accounts itself, or administrators can configure it to let people sign on using third-party credentials. You can tell what kind of account you have when you try to sign in: if your login ID is your email address, Metabase is managing the account, but if you're using the option to sign in which a third-party identity provider, someone else is responsible for authenticating you.
+
+## If your account is managed by Metabase...
```
Comment (paoliniluis) [inline docs/troubleshooting-guide/cant-log-in.md:23]: I would say here: If your instance does not use any SSO or something like that... as maybe some people won't know what might be managed by Metabase?

File: docs/troubleshooting-guide/cant-log-in.md (hunk 9 above)
Comment (gvwilson) [reply docs/troubleshooting-guide/cant-log-in.md:21]: Hm - @jeff-bruemmer thought this was OK. As a regular user, how can I tell if my credentials are in Metabase or via SSO?

File: docs/troubleshooting-guide/cant-log-in.md (hunk 10 above)
Comment (gvwilson) [reply docs/troubleshooting-guide/cant-log-in.md:49]: done - thanks.

File: docs/troubleshooting-guide/cant-log-in.md (hunk 9 above)
Comment (flamber) [reply docs/troubleshooting-guide/cant-log-in.md]: The user would not know that unless they login. And the admin cannot really see that either depending on how the user was initially created.

File: docs/troubleshooting-guide/cant-log-in.md [hunk 12]
```diff
@@ -0,0 +1,93 @@
...
+
+## If you're logging in via LDAP, is LDAP configured correctly?
+
+**Root cause**: The LDAP connection is not configured correctly.
+
+**Steps to take:**
+
+1.  Go to the **Admin Panel** and choose **Authentication**.
+2.  Make sure that LDAP is enabled
+3.  Make sure Metabase has the correct host, port, and login credentials for your LDAP server. You can test this by logging into LDAP directly using some other application, such as [Apache Directory Studio][ads].
```
Comment (paoliniluis) [inline docs/troubleshooting-guide/cant-log-in.md:88]: this is the link, just for reference: https://directory.apache.org/studio/

**Comment by paoliniluis** [comment]
Food for thought: we have a few Cloud customers with this issue (for some reason they forget about their password just after they create it, or they don't know that store password is not the same as the instance password). Do we include that here?

File: docs/troubleshooting-guide/cant-log-in.md (hunk 9 above)
Comment (gvwilson) [reply docs/troubleshooting-guide/cant-log-in.md]: We should chat...

**Pull request merged**